apps/ledger-live-mobile/src/screens/ReceiveFunds/02-SelectAccount.tsx

```diff
@@ -82,16 +85,16 @@ function ReceiveSelectAccount({
   );
 
   const selectAccount = useCallback(
-    (account: AccountLike) => {
+    (account: AccountLikeEnhanced) => {
       if (currency) {
         track("account_clicked", {
           asset: currency.name,
           page: "Select account to deposit to",
         });
         navigation.navigate(ScreenName.ReceiveConfirmation, {
           ...route.params,
-          accountId: (account as SubAccountEnhanced)?.parentId || account.id,
-          createTokenAccount: (account as SubAccountEnhanced)?.triggerCreateAccount,
+          accountId: (account.type !== "Account" && account?.parentId) || account.id,
+          createTokenAccount: "triggerCreateAccount" in account && account?.triggerCreateAccount,
```

Comment:
I don't think that we need to test it twice, am I missing something?
```suggestion
          createTokenAccount: "triggerCreateAccount" in account,
```